common/http.go

```diff
@@ -316,11 +313,20 @@ func getURL(baseURL string, urlString string) (string, error) {
 }
 
 // addHeaders adds the given headers to the request.
-func addHeaders(req *http.Request, headers []Header) (*http.Request, error) {
+func addHeaders(req *http.Request, headers []Header) *http.Request {
 	// Apply any custom headers
 	for _, hdr := range headers {
 		req.Header.Add(hdr.Key, hdr.Value)
 	}
 
-	return req, nil
+	return req
+}
+
+// addContentTypeIfNotPresent adds the Content-Type header if it is not already present.
+func addContentTypeIfNotPresent(req *http.Request) *http.Request {
```

Comment:
Can you rename to `addJSONContentTypeIfNotPresent` or `ensureJSONHeader`? (Basically add JSON to the name so it's obvious what the behavior is.)